api/src/main/java/org/openmrs/module/clientregistry/api/ClientRegistryManager.java

```diff
@@ -82,4 +91,30 @@ public void disableClientRegistry() {
 		
 		isRunning.set(false);
 	}
+	
+	/**
+	 * Determine the appropriate PatientService class based off of the client registry transaction
+	 * type configuration
+	 * 
+	 * @return PatientService class corresponding to the appropriate transaction type supported by
+	 *         the client registry
+	 * @throws IllegalArgumentException if defined transaction type is unsupported
+	 */
+	public CRPatientService getPatientService() throws IllegalArgumentException {
```

Comment:
this only needs to be determined once on module startup, maybe we move the logic into `ClientRegistryActivator` ?